src/radiation_system.hpp

```diff
@@ -1241,12 +1247,24 @@ void RadSystem<problem_t>::AddSourceTerms(array_t &consVar, arrayconst_t &radEne
 				auto fz = Fz / (c_light_ * erad);
 
 				std::array<std::array<double, numRadVars_>, 3> P{};
-				double S = NAN;
-				// loop DIR over {FluxDir::X1, FluxDir::X2, FluxDir::X3}
-				// ComputeRadPressure will modify P[DIR] and S
-				ComputeRadPressure<FluxDir::X1>(P[0], S, erad, Fx, Fy, Fz, fx, fy, fz);
-				ComputeRadPressure<FluxDir::X2>(P[1], S, erad, Fx, Fy, Fz, fx, fy, fz);
-				ComputeRadPressure<FluxDir::X3>(P[2], S, erad, Fx, Fy, Fz, fx, fy, fz);
+				{
+					auto [F, S] = ComputeRadPressure<FluxDir::X1>(erad, Fx, Fy, Fz, fx, fy, fz);
```

Comment:
does it work if you assign it directly to `P[0]`? something like `auto [P[0], S] = ...`?

Reply:
It does not work. Structured bindings do not allow array elements. However, `P[0] = F` works and is simpler.